Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs

```diff
@@ -258,6 +258,18 @@ static class Mono
 		/// </summary>
 		static class CoreCLR
 		{
+#if NET5_0
+			static CoreCLR() =>  NativeLibrary.SetDllImportResolver(typeof(CoreCLR).Assembly,
+				(string libraryName, Assembly assembly, DllImportSearchPath? searchPath) => {
+					if (libraryName == "libcoreclr.so") {
+						var module = typeof(CoreCLR).Module;
+						return Marshal.GetHINSTANCE(module);
```

Comment:
`Marshal.GetHINSTANCE` is just an obscure way to say `(IntPtr)(-1)` outside Windows. Would it be better to just use -1 to make this more explicit?

https://github.com/dotnet/runtime/blob/e7204f5d6fcaca5e097ec854b3be6055229fc442/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.NoCom.cs#L103

Reply:
(We are going to add an analyzer to mark single-file unfriendly APIs and GetHINSTANCE is one of such APIs because on Windows, the behavior of the API is going to change once you turn on single-file. On Linux it won't change anything.)

https://github.com/dotnet/runtime/issues/38405

Reply:
Neat! simplified the resolver. :) I did not realize that GetHINSTANCE simply returns -1 for Unix from GetINSTANCE. I adapted the algorithm in response to Mono's `__Internal` magic string request for self-referencing binary presented here: https://github.com/dotnet/runtime/issues/7267#issuecomment-613516993 (which is a cross-platform solution and here we are only concerned about Linux).

Explicit support of `__Internal` or equivalent would be a nice addition in 6.0 and it can potentially simplify the code in this file. However, as long as we need 5.0 support, we would need the custom resolver here.